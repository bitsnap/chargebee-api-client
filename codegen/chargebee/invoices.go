package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListInvoices generates chargebee client code to fetch all invoices
//
// API: https://{site}.
//
// Documentation: https://apidocs.chargebee.com/docs/api/invoices?lang=curl#list_invoices
func GenerateListInvoices() string {
	return client.GenerateList("Invoices", "Invoice", "chargebee.com/api/v2/invoices")
}

// GenerateRetrieveInvoice generates chargebee client code to retrieve specific invoice
//
// API: https://{site}.chargebee.com/api/v2/invoices/{invoice-id}
//
// Documentation: https://apidocs.chargebee.com/docs/api/invoices?lang=curl#retrieve_an_invoice
func GenerateRetrieveInvoice() string {
	return client.GenerateRetrieve("Invoice", "chargebee.com/api/v2/invoices")
}

// GenerateInvoicesModel generates chargebee Invoice domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/invoices?lang=curl#invoice_attributes
func GenerateInvoicesModel() string {
	return client.GenerateModel("Invoice", "Invoice attributes", "https://apidocs.chargebee.com/docs/api/invoices")
}
