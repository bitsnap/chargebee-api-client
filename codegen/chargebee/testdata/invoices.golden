# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee invoices code generation when generating Chargebee API client should be able to generate invoice models" = '''
(string) (len=10599) "type Invoice struct {\n    Id string `json:\"id\" validate:\"required\"`\n    PoNumber string `json:\"po_number\"`\n    CustomerId string `json:\"customer_id\" validate:\"required\"`\n    SubscriptionId string `json:\"subscription_id\"`\n    Recurring bool `json:\"recurring\" validate:\"required\"`\n    Status enums.StatusEnum `json:\"status\" validate:\"required\"`\n    VatNumber string `json:\"vat_number\"`\n    PriceType enums.PriceTypeEnum `json:\"price_type\" validate:\"required\"`\n    Date *time.Time `json:\"date\"`\n    DueDate *time.Time `json:\"due_date\"`\n    NetTermDays int `json:\"net_term_days\"`\n    ExchangeRate *big.Float `json:\"exchange_rate\"`\n    CurrencyCode string `json:\"currency_code\" validate:\"required\"`\n    Total uint64 `json:\"total\"`\n    AmountPaid uint64 `json:\"amount_paid\"`\n    AmountAdjusted uint64 `json:\"amount_adjusted\"`\n    WriteOffAmount uint64 `json:\"write_off_amount\"`\n    CreditsApplied uint64 `json:\"credits_applied\"`\n    AmountDue uint64 `json:\"amount_due\"`\n    PaidAt *time.Time `json:\"paid_at\"`\n    DunningStatus enums.DunningStatusEnum `json:\"dunning_status\"`\n    NextRetryAt *time.Time `json:\"next_retry_at\"`\n    VoidedAt *time.Time `json:\"voided_at\"`\n    ResourceVersion int64 `json:\"resource_version\"`\n    UpdatedAt *time.Time `json:\"updated_at\"`\n    SubTotal uint64 `json:\"sub_total\" validate:\"required\"`\n    SubTotalInLocalCurrency uint64 `json:\"sub_total_in_local_currency\"`\n    TotalInLocalCurrency uint64 `json:\"total_in_local_currency\"`\n    LocalCurrencyCode string `json:\"local_currency_code\"`\n    Tax uint64 `json:\"tax\" validate:\"required\"`\n    LocalCurrencyExchangeRate *big.Float `json:\"local_currency_exchange_rate\"`\n    FirstInvoice bool `json:\"first_invoice\"`\n    NewSalesAmount uint64 `json:\"new_sales_amount\"`\n    HasAdvanceCharges bool `json:\"has_advance_charges\"`\n    TermFinalized bool `json:\"term_finalized\" validate:\"required\"`\n    IsGifted bool `json:\"is_gifted\" validate:\"required\"`\n    GeneratedAt *time.Time `json:\"generated_at\"`\n    ExpectedPaymentDate *time.Time `json:\"expected_payment_date\"`\n    AmountToCollect uint64 `json:\"amount_to_collect\"`\n    RoundOffAmount uint64 `json:\"round_off_amount\"`\n    PaymentOwner string `json:\"payment_owner\"`\n    VoidReasonCode string `json:\"void_reason_code\"`\n    Deleted bool `json:\"deleted\" validate:\"required\"`\n    TaxCategory string `json:\"tax_category\"`\n    VatNumberPrefix string `json:\"vat_number_prefix\"`\n    Channel enums.ChannelEnum `json:\"channel\"`\n    BusinessEntityId string `json:\"business_entity_id\"`\n    LineItems []LineItem `json:\"line_items\"`\n    Id string `json:\"id\"`\n    SubscriptionId string `json:\"subscription_id\"`\n    DateFrom *time.Time `json:\"date_from\" validate:\"required\"`\n    DateTo *time.Time `json:\"date_to\" validate:\"required\"`\n    UnitAmount uint64 `json:\"unit_amount\" validate:\"required\"`\n    Quantity int `json:\"quantity\"`\n    Amount uint64 `json:\"amount\"`\n    PricingModel enums.PricingModelEnum `json:\"pricing_model\"`\n    IsTaxed bool `json:\"is_taxed\" validate:\"required\"`\n    TaxAmount uint64 `json:\"tax_amount\"`\n    UnitAmountInDecimal string `json:\"unit_amount_in_decimal\"`\n    QuantityInDecimal string `json:\"quantity_in_decimal\"`\n    AmountInDecimal string `json:\"amount_in_decimal\"`\n    DiscountAmount uint64 `json:\"discount_amount\"`\n    ItemLevelDiscountAmount uint64 `json:\"item_level_discount_amount\"`\n    ReferenceLineItemId string `json:\"reference_line_item_id\"`\n    Description string `json:\"description\" validate:\"required\"`\n    EntityDescription string `json:\"entity_description\"`\n    EntityType enums.EntityTypeEnum `json:\"entity_type\" validate:\"required\"`\n    TaxExemptReason enums.TaxExemptReasonEnum `json:\"tax_exempt_reason\"`\n    EntityId string `json:\"entity_id\"`\n    CustomerId string `json:\"customer_id\"`\n    Discounts []Discount `json:\"discounts\"`\n    Amount uint64 `json:\"amount\" validate:\"required\"`\n    Description string `json:\"description\"`\n    LineItemId string `json:\"line_item_id\"`\n    EntityType enums.EntityTypeEnum `json:\"entity_type\" validate:\"required\"`\n    DiscountType enums.DiscountTypeEnum `json:\"discount_type\"`\n    EntityId string `json:\"entity_id\"`\n    CouponSetCode string `json:\"coupon_set_code\"`\n    LineItemDiscounts []LineItemDiscount `json:\"line_item_discounts\"`\n    LineItemId string `json:\"line_item_id\" validate:\"required\"`\n    DiscountType enums.DiscountTypeEnum `json:\"discount_type\" validate:\"required\"`\n    EntityId string `json:\"entity_id\"`\n    DiscountAmount uint64 `json:\"discount_amount\" validate:\"required\"`\n    Taxes []Tax `json:\"taxes\"`\n    Name string `json:\"name\" validate:\"required\"`\n    Amount uint64 `json:\"amount\" validate:\"required\"`\n    Description string `json:\"description\"`\n    LineItemTaxes []LineItemTax `json:\"line_item_taxes\"`\n    LineItemId string `json:\"line_item_id\"`\n    TaxName string `json:\"tax_name\" validate:\"required\"`\n    DateTo *time.Time `json:\"date_to\"`\n    DateFrom *time.Time `json:\"date_from\"`\n    ProratedTaxableAmount *big.Float `json:\"prorated_taxable_amount\"`\n    IsPartialTaxApplied bool `json:\"is_partial_tax_applied\"`\n    IsNonComplianceTax bool `json:\"is_non_compliance_tax\"`\n    TaxableAmount uint64 `json:\"taxable_amount\" validate:\"required\"`\n    TaxAmount uint64 `json:\"tax_amount\" validate:\"required\"`\n    TaxJurisType enums.TaxJurisTypeEnum `json:\"tax_juris_type\"`\n    TaxJurisName string `json:\"tax_juris_name\"`\n    TaxJurisCode string `json:\"tax_juris_code\"`\n    TaxAmountInLocalCurrency uint64 `json:\"tax_amount_in_local_currency\"`\n    LocalCurrencyCode string `json:\"local_currency_code\"`\n    LineItemTiers []LineItemTier `json:\"line_item_tiers\"`\n    LineItemId string `json:\"line_item_id\"`\n    StartingUnit int `json:\"starting_unit\" validate:\"required\"`\n    EndingUnit int `json:\"ending_unit\"`\n    QuantityUsed int `json:\"quantity_used\" validate:\"required\"`\n    UnitAmount uint64 `json:\"unit_amount\" validate:\"required\"`\n    StartingUnitInDecimal string `json:\"starting_unit_in_decimal\"`\n    EndingUnitInDecimal string `json:\"ending_unit_in_decimal\"`\n    QuantityUsedInDecimal string `json:\"quantity_used_in_decimal\"`\n    UnitAmountInDecimal string `json:\"unit_amount_in_decimal\"`\n    LinkedPayments []InvoiceTransaction `json:\"linked_payments\"`\n    TxnId string `json:\"txn_id\" validate:\"required\"`\n    AppliedAmount uint64 `json:\"applied_amount\" validate:\"required\"`\n    AppliedAt *time.Time `json:\"applied_at\" validate:\"required\"`\n    TxnStatus enums.TxnStatusEnum `json:\"txn_status\"`\n    TxnDate *time.Time `json:\"txn_date\"`\n    TxnAmount uint64 `json:\"txn_amount\"`\n    DunningAttempts []DunningAttempt `json:\"dunning_attempts\"`\n    Attempt int `json:\"attempt\" validate:\"required\"`\n    TransactionId string `json:\"transaction_id\"`\n    DunningType enums.DunningTypeEnum `json:\"dunning_type\" validate:\"required\"`\n    CreatedAt *time.Time `json:\"created_at\"`\n    TxnStatus enums.TxnStatusEnum `json:\"txn_status\"`\n    TxnAmount uint64 `json:\"txn_amount\"`\n    AppliedCredits []AppliedCredit `json:\"applied_credits\"`\n    CnId string `json:\"cn_id\" validate:\"required\"`\n    AppliedAmount uint64 `json:\"applied_amount\" validate:\"required\"`\n    AppliedAt *time.Time `json:\"applied_at\" validate:\"required\"`\n    CnReasonCode enums.CnReasonCodeEnum `json:\"cn_reason_code\"`\n    CnCreateReasonCode string `json:\"cn_create_reason_code\"`\n    CnDate *time.Time `json:\"cn_date\"`\n    CnStatus enums.CnStatusEnum `json:\"cn_status\" validate:\"required\"`\n    AdjustmentCreditNotes []CreatedCreditNote `json:\"adjustment_credit_notes\"`\n    CnId string `json:\"cn_id\" validate:\"required\"`\n    CnReasonCode enums.CnReasonCodeEnum `json:\"cn_reason_code\"`\n    CnCreateReasonCode string `json:\"cn_create_reason_code\"`\n    CnDate *time.Time `json:\"cn_date\"`\n    CnTotal uint64 `json:\"cn_total\"`\n    CnStatus enums.CnStatusEnum `json:\"cn_status\" validate:\"required\"`\n    IssuedCreditNotes []CreatedCreditNote `json:\"issued_credit_notes\"`\n    CnId string `json:\"cn_id\" validate:\"required\"`\n    CnReasonCode enums.CnReasonCodeEnum `json:\"cn_reason_code\"`\n    CnCreateReasonCode string `json:\"cn_create_reason_code\"`\n    CnDate *time.Time `json:\"cn_date\"`\n    CnTotal uint64 `json:\"cn_total\"`\n    CnStatus enums.CnStatusEnum `json:\"cn_status\" validate:\"required\"`\n    LinkedOrders []LinkedOrder `json:\"linked_orders\"`\n    Id string `json:\"id\" validate:\"required\"`\n    DocumentNumber string `json:\"document_number\"`\n    Status enums.StatusEnum `json:\"status\"`\n    OrderType enums.OrderTypeEnum `json:\"order_type\"`\n    ReferenceId string `json:\"reference_id\"`\n    FulfillmentStatus string `json:\"fulfillment_status\"`\n    BatchId string `json:\"batch_id\"`\n    CreatedAt *time.Time `json:\"created_at\" validate:\"required\"`\n    Notes []Note `json:\"notes\"`\n    EntityType enums.EntityTypeEnum `json:\"entity_type\" validate:\"required\"`\n    Note string `json:\"note\" validate:\"required\"`\n    EntityId string `json:\"entity_id\"`\n    FirstName string `json:\"first_name\"`\n    LastName string `json:\"last_name\"`\n    Email string `json:\"email\"`\n    Company string `json:\"company\"`\n    Phone string `json:\"phone\"`\n    Line1 string `json:\"line1\"`\n    Line2 string `json:\"line2\"`\n    Line3 string `json:\"line3\"`\n    City string `json:\"city\"`\n    StateCode string `json:\"state_code\"`\n    State string `json:\"state\"`\n    Country string `json:\"country\"`\n    Zip string `json:\"zip\"`\n    ValidationStatus enums.ValidationStatusEnum `json:\"validation_status\"`\n    Index int `json:\"index\" validate:\"required\"`\n    Id string `json:\"id\" validate:\"required\"`\n    Descriptor string `json:\"descriptor\"`\n    FirstName string `json:\"first_name\"`\n    LastName string `json:\"last_name\"`\n    Email string `json:\"email\"`\n    Company string `json:\"company\"`\n    Phone string `json:\"phone\"`\n    Line1 string `json:\"line1\"`\n    Line2 string `json:\"line2\"`\n    Line3 string `json:\"line3\"`\n    City string `json:\"city\"`\n    StateCode string `json:\"state_code\"`\n    State string `json:\"state\"`\n    Country string `json:\"country\"`\n    Zip string `json:\"zip\"`\n    ValidationStatus enums.ValidationStatusEnum `json:\"validation_status\"`\n    Id string `json:\"id\" validate:\"required\"`\n    ReferenceNumber string `json:\"reference_number\"`\n    Status enums.StatusEnum `json:\"status\" validate:\"required\"`\n    Message string `json:\"message\"`\n    LinkedTaxesWithheld []LinkedTaxWithheld `json:\"linked_taxes_withheld\"`\n    Id string `json:\"id\" validate:\"required\"`\n    Amount uint64 `json:\"amount\"`\n    Description string `json:\"description\"`\n    Date *time.Time `json:\"date\"`\n    ReferenceNumber string `json:\"reference_number\"`\n    Timezone string `json:\"timezone\"`\n    Country string `json:\"country\"`\n    RegistrationNumber string `json:\"registration_number\"`\n}"
'''
